sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py

```diff
@@ -24,7 +24,12 @@ class StorageStreamDownloader(object):
     def __init__(self, downloader):
         self._downloader = downloader
         self.name = self._downloader.name
-        self.properties = from_blob_properties(self._downloader.properties)  # pylint: disable=protected-access
+
+        # Parse additional Datalake-only properties
+        additional_args = dict()
+        additional_args['encryption_context'] = self._downloader._response.response.headers.get('x-ms-encryption-context')  # pylint: disable=line-too-long,protected-access
```

Comment:
Just want to call out that I went with `additional_args`. Am happy to take any suggestions!

Reply:
`from_blob_properties` looks good but since you used the `**` approach, you can pass things in like kwargs. So you should be able to do this:
```py
encryption_context = self._downloader._response.response.headers.get('x-ms-encryption-context')

self.properties = from_blob_properties(self._downloader.properties, encryption_context=encryption_context)
```

Just saves creating the extra `dict`.

Reply:
Ohh gotcha. Yes that makes sense!